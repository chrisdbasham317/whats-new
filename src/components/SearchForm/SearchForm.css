input {
  background: #141527;
  border: none;
  color: grey;
  font-size: 20px;
  margin-top: 20px;
  width: 200%;
}

button {
  background: #B569F0;
  border: none;
  border-radius: 30px;
  color: #ffffff;
  font-size: 25px;
  height: 7%;
  position: absolute;
  right: 15px;
  top: 15px;
  width: 20%;
}