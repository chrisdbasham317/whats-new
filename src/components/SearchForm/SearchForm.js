import React, { Component }from 'react';
import './SearchForm.css';

class SearchForm extends Component{
  constructor() {
    super();
    this.state = {
      searchTerm: '',
    }
  }

  handleChange = event => {
    this.setState({ searchTerm: event.target.value });
  }

  render() {
    return (
      <form>
        <input
          type='text'
          placeholder='Search for news articles here'
          name='searchBar'
          value={this.state.searchTerm}
          onChange={event => this.handleChange(event)}
        />
        <button onClick={event => this.props.searchArticles(this.state.searchTerm, event)}>Search Now</button>
      </form>
    )
  }
}

export default SearchForm;