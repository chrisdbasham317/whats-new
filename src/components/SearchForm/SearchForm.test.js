import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  let wrapper
  const mockSearchArticles = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<SearchForm searchArticles={mockSearchArticles} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})