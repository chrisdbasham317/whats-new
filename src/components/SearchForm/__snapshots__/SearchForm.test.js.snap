// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form>
  <input
    name="searchBar"
    onChange={[Function]}
    placeholder="Search for news articles here"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search Now
  </button>
</form>
`;
